torch/_dynamo/variables/misc.py

```diff
@@ -914,6 +914,17 @@ def wraps(fn):
             return variables.functions.FunctoolsPartialVariable(
                 fn, args=rest_args, keywords=kwargs, **options
             )
+        elif self.value is itertools.repeat:
+            from .builder import SourcelessBuilder
+
+            if len(args) < 2:
+                # We cannot risk infinite generator being consumed to exhaustion by dynamo
+                # (i.e. infinite loop)
```

Comment:
How come? Shouldn't this be triggering the generator protocol?

Reply:
Hmm, for some reason, it hangs.

I implemented it as:
```python
if count is None:
  while True:
    yield item
```

Reply:
The reason is that this does not get wrapped into a generator, but rather a `ListIteratorVariable` for some reason.

Reply:
In particular, InliningGeneratorTranslator simply yields the entire generator into a list, which is then returned as a `ListIteratorVariable`:
https://github.com/pytorch/pytorch/blob/8bc04f46fe8e69188fa46f1611b46788a7d4824d/torch/_dynamo/symbolic_convert.py#L2518